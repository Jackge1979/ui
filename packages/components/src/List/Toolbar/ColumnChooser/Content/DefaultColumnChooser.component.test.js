import React from 'react';
import { shallow } from 'enzyme';
import {
	DefaultBody,
	DefaultFooter,
	DefaultHeader,
	getColumnDisplay,
} from './DefaultColumnChooser.components';

const t = jest.fn((_, translationValue) => translationValue.defaultValue);

describe('DefaultHeader', () => {
	it('should render', () => {
		// given
		const props = {
			t,
		};
		// when
		const wrapper = shallow(<DefaultHeader {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});

describe('DefaultBody', () => {
	it('should render', () => {
		// given
		const props = {
			columns: [
				{
					label: 'col1',
					order: 1,
					locked: false,
					hidden: false,
				},
				{
					label: 'col2',
					order: 1,
					locked: true,
					hidden: false,
				},
			],
			onBlurOrder: () => jest.fn(),
			onChangeVisibility: () => jest.fn(),
			onDragAndDrop: () => jest.fn(),
			onKeyPressOrder: () => jest.fn(),
			t,
		};
		// when
		const wrapper = shallow(<DefaultBody {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});

describe('DefaultFooter', () => {
	it('should render', () => {
		// given
		const props = {
			selectAllValue: false,
			onSelectAll: jest.fn(),
			submit: jest.fn(),
			t,
		};
		// when
		const wrapper = shallow(<DefaultFooter {...props} />);
		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});

describe('getColumnDisplay', () => {
	const length = 10;
	const onChangeVisibility = () => jest.fn();
	const onDragAndDrop = () => jest.fn();
	const onBlurOrder = () => jest.fn();
	const onKeyPressOrder = () => jest.fn();
	it('should return a draggrable column displayer', () => {
		// given
		const column = {
			label: 'col1',
			order: 1,
			locked: false,
			hidden: false,
		};
		const index = 0;
		// when
		const ret = getColumnDisplay(
			length,
			onChangeVisibility,
			onDragAndDrop,
			onBlurOrder,
			onKeyPressOrder,
			t,
		)(column, index);
		// then
		expect(ret).toMatchSnapshot();
	});
	it('should return a simple column displayer', () => {
		// given
		const column = {
			label: 'col2',
			order: 1,
			locked: true,
			hidden: false,
		};
		const index = 1;
		// when
		const ret = getColumnDisplay(
			length,
			onChangeVisibility,
			onDragAndDrop,
			onBlurOrder,
			onKeyPressOrder,
			t,
		)(column, index);
		// then
		expect(ret).toMatchSnapshot();
	});
});
