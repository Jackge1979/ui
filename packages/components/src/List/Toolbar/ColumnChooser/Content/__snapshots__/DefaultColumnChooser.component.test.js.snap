// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultBody should render 1`] = `
<DragDropContextProvider
  backend={[Function]}
>
  <div
    className="theme-tc-column-chooser-body tc-column-chooser-body"
    id="column-chooser-content"
  >
    <React.Fragment>
      <DropTarget(DragSource(ColumnDisplayerDraggable))
        hidden={false}
        label="col1"
        length={2}
        locked={false}
        onBlurOrder={[MockFunction]}
        onChangeVisibility={[MockFunction]}
        onDragAndDrop={[MockFunction]}
        onKeyPressOrder={[MockFunction]}
        order={1}
      />
    </React.Fragment>
    <React.Fragment>
      <ColumnDisplayer
        hidden={false}
        label="col2"
        length={2}
        locked={true}
        onBlurOrder={[MockFunction]}
        onChangeVisibility={[MockFunction]}
        onDragAndDrop={[MockFunction]}
        onKeyPressOrder={[MockFunction]}
        order={1}
      />
    </React.Fragment>
  </div>
</DragDropContextProvider>
`;

exports[`DefaultFooter should render 1`] = `
<div
  className="theme-tc-column-chooser-footer tc-column-chooser-footer"
>
  <span
    className="theme-tc-column-chooser-footer-select-all tc-column-chooser-footer-select-all"
  >
    <span>
      <input
        aria-label="select all"
        checked={false}
        id="select-all-checkbox"
        name="selectAll"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
    </span>
    <label
      htmlFor="selectAll"
      id="select-all-label"
    >
      Select All
    </label>
  </span>
  <Translate(ActionButton)
    onClick={[Function]}
  />
</div>
`;

exports[`DefaultHeader should render 1`] = `
<div
  className="theme-tc-column-chooser-header tc-column-chooser-header"
/>
`;

exports[`getColumnDisplay should return a draggrable column displayer 1`] = `
<React.Fragment>
  <DropTarget(DragSource(ColumnDisplayerDraggable))
    hidden={false}
    label="col1"
    length={10}
    locked={false}
    onBlurOrder={[MockFunction]}
    onChangeVisibility={[MockFunction]}
    onDragAndDrop={[MockFunction]}
    onKeyPressOrder={[MockFunction]}
    order={1}
  />
</React.Fragment>
`;

exports[`getColumnDisplay should return a simple column displayer 1`] = `
<React.Fragment>
  <ColumnDisplayer
    hidden={false}
    label="col2"
    length={10}
    locked={true}
    onBlurOrder={[MockFunction]}
    onChangeVisibility={[MockFunction]}
    onDragAndDrop={[MockFunction]}
    onKeyPressOrder={[MockFunction]}
    order={1}
  />
</React.Fragment>
`;
